tf_encrypted/__init__.py

```diff
@@ -46,6 +46,7 @@ def set_protocol(prot: Optional[protocol.Protocol] = None) -> None:
       globals()[name] = func
 
   # record new protocol
+  prot.clear_initializers()
```

Comment:
Run this function every time the protocol changes created a weird side effect where the tests would fail because of ops not belonging to the same graph. This is caused `inspect.getmembers(prot, predicate=inspect.ismethod)` function call. When it iterates through all the members of the protocol it access `.initializers` which is a property. Accessing it like this actually makes it so it gets called. Seems like weird behaviour to me. Clearing the initializers here fixes this issue but I'm not sure it's actually what we want. I feel like the real fix is the `.initializers` property not getting called but not sure there is anything we do about that.

Ideas are welcome.

Reply:
yeah I don't think this is want we want, since we might loose initializers this way.

Reply:
Think we'll need a different way of exposing the functions globally than what we're currently doing.